cloudimg/aws.py

```diff
@@ -485,6 +484,7 @@ def publish(self, metadata):
             snapshot = self.get_snapshot_by_name(metadata.snapshot_name)
 
             if not snapshot:
+                metadata.object_name = metadata.object_name.rstrip(".xz")
```

Comment:
Hmm, that's still not what I meant.

I think this is worse than before because it means that 'publish' method has a weird hidden side-effect of changing its input in a significant way, though only in certain code paths.  That kind of thing leads to bugs, as when looking at code like this:

```python
metadata = get_metadata_somehow()
publisher.publish(metadata)
```

...nobody is going to expect that some core fields on `metadata` now may or may not be different from what they were before.

Imagine for example this scenario:

- a task reaches this part of the code and successfully uploads object with name 'foo'
- later code looks at `metadata` and uses object_name for something, and sees the value 'foo'
- then the task fails for whatever reason
- someone resubmits the task (this is the usual strategy if something fails)
- this time, when we get to `publish`, as the image is already present, it goes into the "Image already exists" path and doesn't patch object_name
- later code looks at `metadata` and uses object_name for something, and sees the value 'foo.xz'

So now we have code which thinks the object name is 'foo' on the first push and thinks it's 'foo.xz' on the retry.  That's probably going to cause some issues.  Code shouldn't have side-effects of rewriting their input like this.